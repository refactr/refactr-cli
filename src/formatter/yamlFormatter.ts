import { safeDump } from 'js-yaml';

export default function yamlFormatter(input): string {
  return safeDump(input, {
    indent: 4,
  });
}
