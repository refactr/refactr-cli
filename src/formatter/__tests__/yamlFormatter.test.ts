import yamlFormatter from '../yamlFormatter';

describe('yamlFormatter', () => {
  test('formats data as YAML', () => {
    // Number
    expect(yamlFormatter(0, [])).toMatchSnapshot();

    // String
    expect(yamlFormatter('foobarbaz', [])).toMatchSnapshot();

    // Boolean
    expect(yamlFormatter(true, [])).toMatchSnapshot();
    expect(yamlFormatter(false, [])).toMatchSnapshot();

    // Array
    expect(
      yamlFormatter(
        [
          {
            a: 1,
            b: 2,
            c: 3
          },
          { a: 4, b: 5, c: 6 }
        ],
        ['a', 'b', 'c']
      )
    ).toMatchSnapshot();

    // Object
    expect(
      yamlFormatter(
        {
          a: 1,
          b: 2,
          c: 3
        },
        ['a', 'b', 'c']
      )
    ).toMatchSnapshot();
  });

  // TODO(.): should we implement field omitting functionality for YAML formatter?
  test.skip('omits data fields if not presented in `fields` argument', () => {
    expect(
      yamlFormatter([{ foo: 'foo', bar: 'bar', baz: 'baz' }], ['foo', 'bar'])
    ).toMatchSnapshot();
  });
});
