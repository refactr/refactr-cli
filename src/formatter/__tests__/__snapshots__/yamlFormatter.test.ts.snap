// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yamlFormatter formats data as YAML 1`] = `
"0
"
`;

exports[`yamlFormatter formats data as YAML 2`] = `
"foobarbaz
"
`;

exports[`yamlFormatter formats data as YAML 3`] = `
"true
"
`;

exports[`yamlFormatter formats data as YAML 4`] = `
"false
"
`;

exports[`yamlFormatter formats data as YAML 5`] = `
"-
    a: 1
    b: 2
    c: 3
-
    a: 4
    b: 5
    c: 6
"
`;

exports[`yamlFormatter formats data as YAML 6`] = `
"a: 1
b: 2
c: 3
"
`;
